### PR TITLE
ARSN-175 Restores old behaviors of errors

### PR DESCRIPTION
**Pull request by ghivert**

### CODE REVIEW

File: lib/errors/index.ts [hunk 1]
```diff
@@ -16,6 +16,10 @@ const isBase = Object.fromEntries(
     Object.keys(rawErrors).map(key => [key, false])
 ) as Is;
 
+// This allows to conditionally add the old behavior of errors to properly
+//   test migration. Activate CI tests with MIGRATED=true yarn test.
+const notMigrated = (process.env.MIGRATED ?? 'false') === 'false'
```
Comment (miniscruff) [inline lib/errors/index.ts]: I feel like migrated is a little too vague. Maybe something like "ErrorIsSupported" or something.

File: lib/network/rpc/utils.ts [hunk 2]
```diff
@@ -55,6 +55,9 @@ export function reconstructError(err: Error) {
     }
 
     const reconstructedErr = new Error(err.message);
+    // This restores the old behavior of errors. This should be removed as soon
```
Comment (miniscruff) [inline lib/network/rpc/utils.ts:58]: We should link a Jira ticket to this so it is easy to spot / search

**Review by miniscruff — CHANGES_REQUESTED** [review]

File: lib/network/rpc/utils.ts (hunk 2 above)
Comment (jbertran) [reply lib/network/rpc/utils.ts:58]: Done below the comment

File: lib/errors/index.ts (hunk 1 above)
Comment (jbertran) [reply lib/errors/index.ts]: Updated. Please check my boolean logic, I'm easily confused on this for some reason :grimacing:

**Review by nicolas2bert — APPROVED** [review]

File: lib/network/rpc/utils.ts [hunk 3]
```diff
@@ -55,6 +55,9 @@ export function reconstructError(err: Error) {
...
+    // as all dependent codebases have been migrated to `is` accessors (ARSN-176).
+    reconstructedErr[err.message] = true;
```
Comment (francoisferrand) [inline lib/network/rpc/utils.ts:60]: should this be conditioned on `allowUnsafeErrComp` ?

File: lib/errors/index.ts (hunk 1 above)
Comment (miniscruff) [reply lib/errors/index.ts]: Looks good now, much easier to make flags like this in the positive term like isX or allowY which might be why you were confused.

**Review by miniscruff — APPROVED** [review]

File: lib/network/rpc/utils.ts (hunk 3 above)
Comment (jbertran) [reply lib/network/rpc/utils.ts:60]: Indeed. Done.

**Pull request merged**